web-local/tests/utils/ResourceWatcher.ts

```diff
@@ -0,0 +1,33 @@
+import { waitUntil } from "@rilldata/web-common/lib/waitUtils";
+import type { Page } from "playwright";
+import { updateCodeEditor } from "web-local/tests/utils/commonHelpers";
+
+const ResourceWatcherLogRegex = /^\[(.*)] rill\.runtime\.v1\.(.*)\/(.*)$/;
+
+export class ResourceWatcher {
+  private statuses = new Map<string, string>();
+
+  public constructor(private readonly page: Page) {
+    page.on("console", (e) => {
+      const matches = ResourceWatcherLogRegex.exec(e.text());
+      if (!matches || matches.length < 4) return;
+      const [, status, , name] = matches;
+      this.statuses.set(name, status);
+    });
+  }
+
+  public async waitForResource(name: string) {
```

Comment:
**Issue**: The method 'waitForResource' uses a hard-coded comparison string "RECONCILE_STATUS_IDLE", which could limit the flexibility of the ResourceWatcher class. <br> **Fix**: Consider making the status string a parameter of the 'waitForResource' method or part of the class configuration to increase flexibility and reusability of the 'ResourceWatcher' class. <br> **Code Suggestion**: 
 ```
 - return waitUntil(() => this.statuses.get(name) === "RECONCILE_STATUS_IDLE");
 + return waitUntil(() => this.statuses.get(name) === this.idleStatus); // this.idleStatus should be configurable
 ```

Reply:
It doesnt make sense to make this configurable.